public/components/trace_analytics/components/app.tsx

```diff
@@ -0,0 +1,158 @@
+import { handleIndicesExistRequest } from '../requests/request_handler';
+import { SearchBarProps } from './common';
+import { FilterType } from './common/filters/filters';
+import { renderPageWithSidebar } from './common/side_nav';
+import { Dashboard } from './dashboard';
+import { Services, ServiceView } from './services';
+import { Traces, TraceView } from './traces';
+
+interface TraceAnalyticsAppDeps {
+  http: CoreStart['http'];
+  chrome: CoreStart['chrome'];
+}
+
+export interface CoreDeps {
+  http: CoreStart['http'];
+  setBreadcrumbs: (newBreadcrumbs: ChromeBreadcrumb[]) => void;
+  indicesExist?: boolean;
+}
+
+export const TraceAnalyticsApp = ({ http, chrome }: TraceAnalyticsAppDeps) => {
+  const [indicesExist, setIndicesExist] = useState(true);
+  const storedFilters = sessionStorage.getItem('TraceAnalyticsFilters');
+  const [query, setQuery] = useState<string>(sessionStorage.getItem('TraceAnalyticsQuery') || '');
+  const [filters, setFilters] = useState<FilterType[]>(
+    storedFilters ? JSON.parse(storedFilters) : []
+  );
+  const [startTime, setStartTime] = useState<string>(
+    sessionStorage.getItem('TraceAnalyticsStartTime') || 'now-5m'
+  );
+  const [endTime, setEndTime] = useState<string>(
+    sessionStorage.getItem('TraceAnalyticsEndTime') || 'now'
+  );
+
+  const setFiltersWithStorage = (newFilters: FilterType[]) => {
+    setFilters(newFilters);
+    sessionStorage.setItem('TraceAnalyticsFilters', JSON.stringify(newFilters));
+  };
+  const setQueryWithStorage = (newQuery: string) => {
+    setQuery(newQuery);
+    sessionStorage.setItem('TraceAnalyticsQuery', newQuery);
+  };
+  const setStartTimeWithStorage = (newStartTime: string) => {
+    setStartTime(newStartTime);
+    sessionStorage.setItem('TraceAnalyticsStartTime', newStartTime);
+  };
+  const setEndTimeWithStorage = (newEndTime: string) => {
+    setEndTime(newEndTime);
+    sessionStorage.setItem('TraceAnalyticsEndTime', newEndTime);
+  };
+
+  useEffect(() => {
+    handleIndicesExistRequest(http, setIndicesExist);
+  }, []);
+
+  const commonProps: SearchBarProps & CoreDeps = {
+    http,
+    uiSettings,
+    setBreadcrumbs: chrome.setBreadcrumbs,
+    query,
+    setQuery: setQueryWithStorage,
+    filters,
+    setFilters: setFiltersWithStorage,
+    startTime,
+    setStartTime: setStartTimeWithStorage,
+    endTime,
+    setEndTime: setEndTimeWithStorage,
+    indicesExist,
+  };
+
+  return (
+    <HashRouter>
+      <I18nProvider>
+        <>
+          <Switch>
+            <Route
+              exact
+              path={['/dashboard', '/']}
+              render={(props) => renderPageWithSidebar(<Dashboard {...commonProps} />, 1)}
+            />
+            <Route
+              exact
+              path="/traces"
+              render={(props) => renderPageWithSidebar(<Traces {...commonProps} />, 2)}
+            />
+            <Route
+              path="/traces/:id+"
```

Comment:
Have you tried these routes with both '/trace_analytics', '/trace_analytics/home' prefixes? For me it was giving some issue for custom_panel_view page taking in param.

Reply:
ah forgot to remove this file, it got moved to `home.tsx` and is not used anymore, thanks for catching this